independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -136,20 +143,35 @@ static void addInterceptedMethodCandidates(BeanDeployment beanDeployment, ClassI
                 if (Modifier.isFinal(method.flags())) {
                     String className = method.declaringClass().name().toString();
                     if (!className.startsWith("java.")) {
-                        LOGGER.warn(
-                                String.format(
-                                        "Method %s.%s() is final, skipped during generation of the corresponding intercepted subclass",
-                                        className, method.name()));
+                        methodsFromWhichToRemoveFinal.add(method.name());
                     }
-                } else {
-                    candidates.computeIfAbsent(new Methods.MethodKey(method), key -> merged);
                 }
+                candidates.computeIfAbsent(new Methods.MethodKey(method), key -> merged);
             }
         }
+        if (!methodsFromWhichToRemoveFinal.isEmpty()) {
+            bytecodeTransformerConsumer.accept(
+                    new BytecodeTransformer(classInfo.name().toString(), new BiFunction<String, ClassVisitor, ClassVisitor>() {
+                        @Override
+                        public ClassVisitor apply(String s, ClassVisitor classVisitor) {
+                            return new ClassVisitor(Opcodes.ASM7, classVisitor) {
+                                @Override
+                                public MethodVisitor visitMethod(int access, String name, String descriptor, String signature,
+                                        String[] exceptions) {
+                                    if (methodsFromWhichToRemoveFinal.contains(name)) {
```

Comment:
Technically this should match the signature as well as there could be multiple methods with the same name. In practice I don't think it will make any real difference though, removing final is not a big deal for what we are doing.

Reply:
I made the same comment above only with a few seconds difference :P

Reply:
This should be fixed now that I lifted some of the utils from gizmo